ci/build.yml

```diff
@@ -17,6 +17,9 @@ build and push image:
     - /^run\//
   script:
     - docker build --tag $CONTAINER_NAME:$CI_COMMIT_SHA --tag $CONTAINER_NAME:latest -f docker/Dockerfile .
+    - docker-compose -f docker/compose.yml up trezor-user-env-unix &
+    - nix-shell --run "poetry install"
+    - nix-shell --run "poetry run make test_with_running_controller"
```

Comment:
I do not like this. I understand the point of it, but we should not test in the build stage, it is confusing when it fails. I would remove it.

Reply:
Yes, I agree, it is not very good. Done by suggested process of using the `image` directive using the freshly-built `tenv` image.

One (small) disadvantage of running tests _inside_ the image - we are _not_ using the image exactly as it is used in production. Let's say `Suite` is downloading the image and running it by `docker-compose` and communicates with exposed ports. Here we are not connecting to the running container at all, so we are not testing the `docker` exposes ports and we do not test the `CMD` instruction that is executed _only_ when the container runs (we need to replicate it by hand in our testing code).

Not sure if that makes sense and if it is even a "problem" :)